classes/local/apibridge.php

```diff
@@ -1791,4 +1849,226 @@ public function get_duplicated_episodeid($workflowid) {
+                    if (empty($existingacl)) {
+                        $acl[] = $roleobject;
+                    }
+                }
+            }
+        }
+
+        // Put everything in params value as a string.
+        $params['acl'] = json_encode(array_values($acl));
+
+        // When there is nothing to change.
+        if ($params['acl'] == ($jsonacl)) {
+            return true;
+        }
+
+        // Update the acls via put request.
+        $api = new api();
+        $api->oc_put($resource, $params);
+
+        // Finally we return the result of that request to the server.
+        return ($api->get_http_code() == 200);
+    }
+
+    /**
+     * Make Event visible for the new course that has been imported to, by changing the ACLs.
+     *
+     * @param string $identifier event identifier
+     * @return int $courseid id of the course being imported to.
+     */
+    private function imported_events_acl_change($identifier, $courseid) {
+        // Use try to catch unwanted errors.
+        try {
+            // Make it visible to the course does the ACL change accordingly.
+            $visibilychanged = $this->change_visibility($identifier, $courseid, block_opencast_renderer::VISIBLE);
+            // In order to resolve the return result of the change_visibilty method, we assume non (false) values as true.
+            return ($visibilychanged !== false) ? true : false;
+        } catch (\moodle_exception $e) {
+            // It is unlikely, but who knows.
+            return false;
+        }
+    }
+
+    /**
+     * Map the seriesid to the course that has been imported to, by assinging the series as secondary.
+     *
+     * @return int $courseid id of the course being imported to.
+     * @param string $seriesid series id.
+     */
+    private function map_imported_series_to_course($courseid, $seriesid) {
+        try {
+            // Get the current record.
+            $mapping = seriesmapping::get_record(array('courseid' => $courseid, 'series' => $seriesid, 'isdefault' => '0'));
+
+            // If the mapping record does not exists, we create one.
+            if (!$mapping) {
+                $mapping = new seriesmapping();
+                $mapping->set('courseid', $courseid);
+                $mapping->set('series', $seriesid);
+                $mapping->set('isdefault', 0); // It is not the default series of this course.
+                $mapping->create();
+            }
+        } catch (\moodle_exception $e) {
+            // Unlikely but who knows.
+            return false;
+        }
+
+        // Return true, except when there is a system error.
+        return true;
+    }
+
+    /**
+     * Retrieves the series data by series identifier.
+     *
+     * @param int $seriesid
+     * @return null|string id of the series id if it exists in the opencast system.
+     */
+    public function get_series_data_by_identifier($seriesid) {
```

Comment:
Is there any reason why don't you use `get_series_by_identifier`? It's doing exactly the same except that it doesn't check the returned http code. However, you could have added this to the old method.
Please use the existing function in order to avoid duplicated code, if possible.